runtime/src/jycessing/LibraryImporter.java

```diff
@@ -137,9 +137,18 @@ protected void addLibrary(final String libName) {
 
     log("mainJar: " + mainJar);
     log("Adding dir: " + contentsDir);
-    recursivelyAddJarsToClasspath(contentsDir);
     log("Platform: " + PLATFORM + " Bits: " + BITS);
-    if (PLATFORM.indexOf("windows") != -1) {
+    if (PLATFORM.equals("macosx")) {
+      log("Looks like we're on a mac");
+      File nativeDir;
+      nativeDir = new File(libDir, "library/macosx");
+      if (!nativeDir.isDirectory()) {
+        nativeDir = contentsDir;
+      }
+      recursivelyLoadNativeLibraries(nativeDir);
+    }
+    if (PLATFORM.equals("windows")) {
+      log("Looks like windows");
```

Comment:
I think these ifs would be better expressed as a switch statement.

Reply:
Yeah, you're absolutely correct. I'll clean that up and do another commit.